### DISCUSSION TITLE
Python bindings?

### DISCUSSION THREAD

**Discussion by richardr1126** [LostRuins/koboldcpp#327] [Ideas]
Now that **Koboldcpp** is one of the best ways to run non-llama models with CuBlas, it is so frustrating having to use **ctransformers** for Python bindings, running on just CPU. Will we ever see a kobold-cpp-python? Or do you think **ctransformers** will get CUDA support sooner than that?

**Comment by LostRuins** [comment]
The python bindings already exist and are usable - although they're more intended for internal use rather than downstream external apps (which are encouraged to use the webapi instead). You can take a look at the `koboldcpp.py` file inside the repo to see how they are being used from the dll.
